pkg/reconciler/utils/authtype/authtype.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authtype
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/google/knative-gcp/pkg/reconciler/identity/resources"
+)
+
+type AuthTypes string
+
+type AuthTypeArgs struct {
+	Namespace          string
+	ServiceAccountName string
+	Secret             *corev1.SecretKeySelector
+}
+
+const (
+	AuthenticationCheckUnknownReason           = "AuthenticationCheckPending"
+	ControlPlaneNamespace                      = "cloud-run-events"
+	BrokerServiceAccountName                   = "broker"
+	Secret                           AuthTypes = "secret"
```

Comment:
The 3 different `AuthTypes` should have GoDoc comments. In particular, the code reader should be able to understand the difference between the two WI options.